modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/DashboardAnonViewTest.java

```diff
@@ -0,0 +1,312 @@
+/*
+*Copyright (c) 2015​, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.ds.ui.integration.test.dashboard;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.*;
+import org.testng.annotations.*;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.ds.integration.common.clients.ResourceAdminServiceClient;
+import org.wso2.ds.ui.integration.util.DSUIIntegrationTest;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test Anonymous Mode for Dashboard with Various Scenarios
+ */
+public class DashboardAnonViewTest extends DSUIIntegrationTest {
+    private static final Log LOG = LogFactory.getLog(DashboardAnonViewTest.class);
+    private static final String DASHBOARD_TITLE = "anondashboard";
+    private static final String DASHBOARD_DESCRIPTION = "This is sample descrition for dashboard";
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+    private String dashboardTitle;
+    private WebElement webElement = null;
+
+    @Factory(dataProvider = "userMode")
+    public DashboardAnonViewTest(TestUserMode userMode, String dashboardTitle) {
+        super(userMode);
+        this.dashboardTitle = dashboardTitle;
+    }
+
+    @DataProvider(name = "userMode")
+    private static Object[][] userModeProvider() {
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN, DASHBOARD_TITLE}};
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setUp() throws Exception {
+        String backendURL = getBackEndUrl();
+        resourceAdminServiceClient = new ResourceAdminServiceClient(backendURL, getCurrentUsername(),
+                getCurrentPassword());
+        resourcePath = DASHBOARD_REGISTRY_BASE_PATH + dashboardTitle.toLowerCase();
+        DSUIIntegrationTest.login(getDriver(), getBaseUrl(), getCurrentUsername(), getCurrentPassword());
+    }
+
+    /**
+     * Test case for adding anon view for the dashboard.
+     */
+    @Test(groups = "wso2.ds.dashboard", description = "Adding anonymous dashboard for dashboard")
+    public void testAnonDashboard() throws Exception {
+        String defaultViewGadgetExpected = "USA Map";
```

Comment:
This is too long testcase, and covers multiple tests. It would be better to break this testcases and make it more readable.